### PR TITLE
:book: Fix docs to support dark-light theme toggling

### PR DESCRIPTION
**Pull request by KPRoche**
Preview at https://kproche.github.io/kubestellar/doc-darklight
## Summary
This PR enables the rendered website to toggle between dark and light modes:
- It adds and activates the toggle icon next to  the search bar
- architecture images with transparent backgrounds have had a white background added
- the small KubeStellar banner image has had its background set to transparent
- the primary (link and menu bar) color is set to light blue in dark mode for better readability
- the accent (link mouseover) color is set to cyan in dark mode
## Related issue(s)
#2386 
Fixes #

### CODE REVIEW

**Comment by MikeSpreitzer** [comment]
Preview at https://kproche.github.io/kubestellar/doc-darklight/direct/architecture/

**Comment by MikeSpreitzer** [comment]
@KPRoche , as you noted in https://github.com/kubestellar/kubestellar/issues/2386#issuecomment-2310823108 :

> BTW, for the record, the main problem was fixed by just adding a style="background-color:white" attribute into the main SVG tag in each file generating a diagram; that keeps them from rendering with a transparent background, which is what causes the visibility issue in dark mode.
> Adding the dark/light toggle and tweaking the blue accent colors for dark mode into the mkdocs.yml are just additional tweaks to improve the user experience.

Do we need/want to take on the responsibility for a toggle in the website itself? There are already multiple controls in the browser (as my journey to see the problem shows) and in the desktop. Is another one really a boon to the user?

**Comment by MikeSpreitzer** [comment]
[taps the sign](https://docs.google.com/document/d/1iwkhpoV76-J-3G-Rif4B1cmL9SWBO_f_fE6WK8jvsDc/edit)

**Comment by KPRoche** [comment]
> [taps the sign](https://docs.google.com/document/d/1iwkhpoV76-J-3G-Rif4B1cmL9SWBO_f_fE6WK8jvsDc)

Would you believe that this time I *did* do a rebase and squashed almost all my commits into one? Honest!

**Comment by KPRoche** [comment]
> Do we need/want to take on the responsibility for a toggle in the website itself? There are already multiple controls in the browser (as my journey to see the problem shows) and in the desktop. Is another one really a boon to the user?

The toggle is a very simple accessibility fix, supported and implemented just by adding about 12 lines to the mkdocs.yml profile.

There are certainly days (if I'm working late for instance) when I appreciate a website letting me flip its color mode. 
Also, given how recalcitrant the browser controls proved to be, and how little visual real estate the toggle consumes, I think it's a net plus. All the hard work is built into the mkdocs engine and materials theme we're using.

**Comment by MikeSpreitzer** [comment]
@KPRoche: good job on the squash. My problem is the other commit, the one that merges `main` into this PR's branch. That is why I tapped the sign.

**Comment by KPRoche** [comment]
> @KPRoche: good job on the squash. My problem is the other commit, the one that merges `main` into this PR's branch. That is why I tapped the sign.
> 
That was because in the approximately 20 minutes between when I pushed my branch back up to github, previewed the results and then could do the PR, you did an update to your go-lang fixes. Because they involved no overlapping files, letting the github.com sync merge those fixes from main back into my branch did not seem like it would break anything.

**Comment by KPRoche** [comment]
Of course, now there even more changes since I did the PR...

**Comment by KPRoche** [comment]
Oooh, the "files changed" preview is particularly handy to show the background changes in the images:
https://github.com/kubestellar/kubestellar/pull/2411/files

File: docs/content/direct/images/kubeflex-architecture.png
Comment (MikeSpreitzer) [inline docs/content/direct/images/kubeflex-architecture.png:1]: Not seeing a visible change for this one.

**Review by MikeSpreitzer — APPROVED** [review]
/lgtm
/approve
Thanks!

File: docs/content/direct/images/kubeflex-architecture.png
Comment (KPRoche) [reply docs/content/direct/images/kubeflex-architecture.png:1]: Most of the PNG files already had a white background. I think I had to fix one of them, or just turn off the transparency bit.

**Pull request merged**